src/components/application_manager/test/state_controller/state_controller_test.cc

```diff
@@ -3428,6 +3464,27 @@ TEST_F(
       media_app_, kDefaultWindowId, new_state, send_activate_app);
 }
 
+TEST_F(
+    StateControllerImplTest,
+    SetRegularState_WEPAppIsInResumingModeEmbeddedNaviIsActive_HmiStateIsNotChanged) {
+  am::event_engine::Event embedded_navi_event(
+      hmi_apis::FunctionID::BasicCommunication_OnEventChanged);
+  smart_objects::SmartObject message;
+  message[am::strings::msg_params][am::hmi_notification::is_active] = true;
+  message[am::strings::msg_params][am::hmi_notification::event_name] =
+      hmi_apis::Common_EventTypes::EMBEDDED_NAVI;
+  embedded_navi_event.set_smart_object(message);
+  state_ctrl_->on_event(embedded_navi_event);
+
+  EXPECT_CALL(*media_wep_app_ptr_, is_resuming()).WillRepeatedly(Return(true));
```

Comment:
@AByzhynar ON_CALL?

Reply:
@mvorobio ON_CALL is not a good idea in this case as I expect this call, the test should fail if this method won't be invoked. But anyway I have made the expectation more precise. 
Please see 35bb27fad6380b3aa53014a01327a1224713c024